crates/assemble/src/lib.rs

```diff
@@ -384,14 +386,10 @@ pub fn journal_selector(
 ) -> broker::LabelSelector {
     let mut include = labels::build_set([
         (labels::COLLECTION, collection.name.as_ref()),
-        // TODO(johnny): Enable this as soon as the label prefix change has propagated
-        // to all connectors, such as derive-typescript.
-        /*
         (
             "name:prefix",
```

Comment:
What's the purpose of this? Seems like this new selector is essentially equivalent to the old one. And aren't they implicitly `AND`ed together anyway? What am I missing?

Reply:
There's a performance optimization in gazette where it uses name:prefix to efficiently seek to the range of possible matches. Same with `id:prefix`. We use both `estuary.dev/collection` and `name:prefix` consistently with this change. I'd held this particular instance back because connectors need to be able to parse it and needed the protocol update.